src/main/java/tetris/TetrisFrame.java

```diff
@@ -0,0 +1,62 @@
+package tetris;
+
+import javax.swing.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+
+//resource: https://www.ssaurel.com/blog/learn-to-create-a-tetris-game-in-java-with-swing/
+//resource: https://docs.oracle.com/javase/7/docs/api/javax/swing/Timer.html
+
+public class TetrisFrame extends JFrame {
+    private TetrisGame tetris;
+    private TetrisView view;
+    private TetrisKeyListener keyListener;
+    private Timer timer;
+
+    public TetrisFrame(TetrisGame tetrisGame, TetrisView tetrisView, TetrisKeyListener tetrisKeyListener){
+        super();
+        tetris = tetrisGame;
+        view = tetrisView;
+        keyListener = tetrisKeyListener;
+
+        setUpFrame();
+        addView();
+        addTimer();
+        tetris.init();
+    }
+
+    private void addView() {
+        add(view);
+    }
+
+    private void setUpFrame() {
+        setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
+        setTitle("Tetris");
+
+        //int frameHeight = (TetrisView.CELL_SIZE * (TetrisGame.HEIGHT - 1)) + TetrisView.BORDER_CELL_SIZE;
+        //int frameWidth = (TetrisView.CELL_SIZE * (TetrisGame.WIDTH - 1)) + TetrisView.BORDER_CELL_SIZE;
+
+        setSize(550, 690);
```

Comment:
I got these numbers by just playing around but there has to be a more logical way to get the size of the view.

Reply:
Hmm Maybe by multiplying dimensions of the view or something? Just a guess. But you're right that there's definitely a way to do it without using arbitrary numbers

Reply:
I did try that in the commented part. But then I just got rid of it cause it was too small and the arbitrary numbers work.